### PR TITLE
First cut of elliptic orbits

### PR DESCRIPTION
**Pull request by eggrobin**
Accessors for other characteristics might be nice.

### CODE REVIEW

File: physics/kepler_orbit.hpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+struct KeplerianElements {
+  double eccentricity;
+  Length semimajor_axis;
+  Angle inclination;
+  Angle longitude_of_ascending_node;
+  Angle argument_of_periapsis;
+  Angle mean_anomaly;
+};
+
+struct TwoBodySystem {
```
Comment (pleroy) [inline physics/kepler_orbit.hpp]: This struct doesn't seem too useful.  I would prefer to pass the two bodies at construction.

File: physics/kepler_orbit.hpp [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  not_null<MassiveBody const*> const primary;
+  not_null<Body const*> const secondary;
+};
+
+template<typename Frame>
+class KeplerOrbit {
+ public:
+  KeplerOrbit(TwoBodySystem const& system,
+              Instant const& epoch,
+              KeplerianElements<Frame> const& elements_at_epoch);
```
Comment (pleroy) [inline physics/kepler_orbit.hpp]: Blank line after.

File: physics/kepler_orbit.hpp [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  RelativeDegreesOfFreedom<Frame> StateVectors(Instant const& t) const;
```
Comment (pleroy) [inline physics/kepler_orbit.hpp]: Same.

File: physics/kepler_orbit_body.hpp [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+KeplerOrbit<Frame>::KeplerOrbit(
+    TwoBodySystem const& system,
+    Instant const& epoch,
+    KeplerianElements<Frame> const& elements_at_epoch)
+    : system_gravitational_parameter_(
+          system.primary->gravitational_parameter() +
+          (system.secondary->is_massless()
+               ? GravitationalParameter{}
+               : CHECK_NOTNULL(
+                     dynamic_cast<MassiveBody const*>(&*system.secondary))->
```
Comment (pleroy) [inline physics/kepler_orbit_body.hpp]: Would `system.secondary.get()` work?

File: physics/kepler_orbit_body.hpp [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+                     mean_anomaly + eccentricity * Radian);
+    Angle const true_anomaly =
+       2 * ArcTan(Sqrt(1 + eccentricity) * Sin(eccentric_anomaly / 2),
+                  Sqrt(1 - eccentricity) * Cos(eccentric_anomaly / 2));
+    Bivector<double, Frame> const x({1, 0, 0});
+    Bivector<double, Frame> const y({0, 1, 0});
+    Bivector<double, Frame> const z({0, 0, 1});
+    // It would be nice to have a local frame, rather than make this a rotation
+    // Frame -> Frame.
+    // TODO(egg): Constructor for |Rotation| using Euler angles.
```
Comment (pleroy) [inline physics/kepler_orbit_body.hpp]: Yes!  Please create an issue.

File: physics/kepler_orbit_body.hpp [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+        (Rotation<Frame, Frame>(Ω, z) *
+         Rotation<Frame, Frame>(i, x) *
+         Rotation<Frame, Frame>(ω, z));
+    Length const distance = a * (1 - eccentricity * Cos(eccentric_anomaly));
+    Displacement<Frame> const r =
+        distance * from_orbit_plane(Vector<double, Frame>({Cos(true_anomaly),
+                                                           Sin(true_anomaly),
+                                                           0}));
+    Velocity<Frame> const v =
+      Sqrt(μ * a) / distance *
```
Comment (pleroy) [inline physics/kepler_orbit_body.hpp]: Bad indent.

File: physics/kepler_orbit_body.hpp [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+          from_orbit_plane(
+                Vector<double, Frame>(
```
Comment (pleroy) [inline physics/kepler_orbit_body.hpp]: Bad indent.

File: physics/kepler_orbit.hpp (hunk 3 above)
Comment (pleroy) [inline physics/kepler_orbit.hpp]: Also the name is not very informative.  How about GetRelativeDegreesOfFreedom?  Also, document relative to what (it's unclear to me why the result is not a normal DoF).

File: physics/kepler_orbit_test.cpp [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+  auto const earth = solar_system.MakeMassiveBody(
+                         solar_system.gravity_model_message("Earth"));
+  auto const moon = solar_system.MakeMassiveBody(
+                        solar_system.gravity_model_message("Moon"));
+  // The numbers in the gravity models and those from the query above both come
+  // from DE431, so the sums are the same up to round-off.
+  EXPECT_THAT(
+      earth->gravitational_parameter() + moon->gravitational_parameter(),
+      AlmostEquals(
+          4.0350323550225975E+05 * Pow<3>(Kilo(Metre)) / Pow<2>(Second), 1));
```
Comment (pleroy) [inline physics/kepler_orbit_test.cpp]: () around the unit to take advantage of constexpr'ness.  Same comment elsewhere in this test.

File: physics/kepler_orbit.hpp (hunk 1 above)
Comment (eggrobin) [inline physics/kepler_orbit.hpp]: Done.

File: physics/kepler_orbit.hpp (hunk 2 above)
Comment (eggrobin) [inline physics/kepler_orbit.hpp]: Doen.

File: physics/kepler_orbit.hpp (hunk 3 above)
Comment (eggrobin) [inline physics/kepler_orbit.hpp]: "state vectors" is a [standard term](https://en.wikipedia.org/wiki/Orbital_state_vectors).
Of one body relative to the other, we don't have much else to relate to.

File: physics/kepler_orbit_body.hpp (hunk 4 above)
Comment (eggrobin) [inline physics/kepler_orbit_body.hpp]: OBE.

File: physics/kepler_orbit_body.hpp (hunk 5 above)
Comment (eggrobin) [inline physics/kepler_orbit_body.hpp]: #810.

File: physics/kepler_orbit_body.hpp (hunk 6 above)
Comment (eggrobin) [inline physics/kepler_orbit_body.hpp]: Done.

File: physics/kepler_orbit_body.hpp (hunk 7 above)
Comment (eggrobin) [inline physics/kepler_orbit_body.hpp]: Done.

File: physics/kepler_orbit_test.cpp (hunk 8 above)
Comment (eggrobin) [inline physics/kepler_orbit_test.cpp]: Done.

**Pull request merged**